src/components/sync-sites-modal-selector.tsx

```diff
@@ -100,6 +100,17 @@ function SearchSites( {
 	);
 }
 
+const getSortedSites = ( sites: SyncSite[] ) => {
+	const order: Record< SyncSite[ 'syncSupport' ], number > = {
+		syncable: 1,
+		'already-connected': 2,
+		'needs-transfer': 3,
+		unsupported: 4,
+	};
+
+	return [ ...sites ].sort( ( a, b ) => order[ a.syncSupport ] - order[ b.syncSupport ] );
```

Comment:
Looks much easier to read. Thanks for simplifying that.

Should we account for a case in which the `syncSypport` key doesn't exist in the object of the order?

Reply:
At least at teh moment, it's redundant, since we always have default value as `syncable` https://github.com/Automattic/studio/blob/trunk/src/hooks/use-fetch-wpcom-sites.tsx#L58

And also, we have the same with TS types - it should always be in the object https://github.com/Automattic/studio/blob/trunk/src/hooks/use-fetch-wpcom-sites.tsx#L6

Also, if we make it optional - we will immediately get an error [here](https://github.com/Automattic/studio/pull/635/files#diff-d03b8b1c01ff73a0f2706d10c3e244ddee8b5a909c4c404952f42907ea4dad0fR104), so we won't miss the case when it becomes optional and as a result won't have a broken UI

Reply:
Sounds good, thanks for confirming that.